staging/src/k8s.io/apiserver/pkg/endpoints/filters/authentication.go

```diff
@@ -154,6 +154,13 @@ func audiencesAreAcceptable(apiAuds, responseAudiences authenticator.Audiences)
 	return len(apiAuds.Intersect(responseAudiences)) > 0
 }
 
+func shouldCloseAnonymousHTTP2Request(req *http.Request, resp *authenticator.Response) bool {
+	return utilfeature.DefaultFeatureGate.Enabled(genericfeatures.UnauthenticatedHTTP2DOSMitigation) &&
+		req.ProtoMajor == 2 &&
+		!resp.IsProxiedAuthentication &&
```

Comment:
I don't have a firm grasp of the network / request layers where the http/2 issues manifested to know if exempting request proxies is ok. Is it possible a proxy could pass through the http/2 constructs that would allow the anonymous end-user to abuse the kube-apiserver, or does the termination / header injection by the proxy mean it *must* be insulating the kube-apiserver from http/2 stuff sent by the end user?

Reply:
> Is it possible a proxy could pass through the http/2 constructs that would allow the anonymous end-user to abuse the kube-apiserver or does the termination / header injection by the proxy mean it _must_ be insulating the kube-apiserver from http/2 stuff sent by the end user?

While the proxy will not pass through the raw constructs, it will pass through the 'work' by making high amounts of requests to the backend (assuming that is the variant of the http/2 attack being used).  This type of work flooding is what Cloudflare saw (i.e. a proxy that was being attacked would flood its backend with work).

My reasoning for this change isn't that the server is being insulated from the attack by the front proxy, but instead that the backend server cannot correctly perform a connection reset based mitigation when the connection is being shared by multiple clients.  We are assuming that the front proxy is mitigating http/2 based attacks in this scenario.

Reply:
> We are assuming that the front proxy is mitigating http/2 based attacks in this scenario.

That makes sense. Do we know if the front proxies we know of are actually mitigating? I guess kube-apiserver → aggregated API server is. Any idea about others like envoy/nginx?

Reply:
I don't know specifics about nginx but envoy definitely does, though it requires rather esoteric config to actually protect the API server effectively.  I know some platforms are successfully using HA proxy as a request header proxy and it is able to mitigate http2 issues as well.

Reply:
Ok. Needs a good release note.